gc/base/ParallelDispatcher.cpp

```diff
@@ -723,20 +723,17 @@ MM_ParallelDispatcher::expandThreadPool(MM_EnvironmentBase *env)
 	Assert_MM_false(_inShutdown);
 
 	Assert_MM_true(_threadShutdownCount == (_threadCountMaximum - 1));
-	Assert_MM_true(_threadCountMaximum == _extensions->gcThreadCount);
 
 	bool result = true;
 
-	uintptr_t supportedGCThreadCount = _extensions->configuration->supportedGCThreadCount(env);
-	uintptr_t preExpandThreadCount = _extensions->gcThreadCount;
-	uintptr_t newThreadCount = OMR_MIN(_threadCountMaximumAtCheckpointStartup, supportedGCThreadCount);
+	uintptr_t preExpandThreadCount = _threadCountMaximum;
+	uintptr_t newThreadCount = _extensions->gcThreadCount;
 
 	Trc_MM_ParallelDispatcher_expandThreadPool_params(
-			newThreadCount, _threadCountMaximumAtCheckpointStartup,
-			supportedGCThreadCount, preExpandThreadCount);
-
-	if (newThreadCount > preExpandThreadCount) {
+			newThreadCount, _poolMaxCapacity,
+			_extensions->configuration->supportedGCThreadCount(env), preExpandThreadCount);
 
+	if ((newThreadCount > preExpandThreadCount) && (result = reinitialize(env, newThreadCount))) {
```

Comment:
do the table reinitialize as an explicit step ahead of this thread count check

Reply:
consider moving the thread count check inside internalStartupThreads - it would be consistent with how/when reinitialize for tables does the check

Reply:
We'll need this check in this method,`internalStartupThreads` does the check already `	while (workerThreadCount < maxWorkerThreadIndex) ` but its return value is based on wether it failed or succeeded in starting the threads independent if check fails or passes. We need the check to set the threadCount vars aside from calling `internalStartupThreads`